pkg/tests/apis/folder/folders_test.go

```diff
@@ -639,3 +644,167 @@ func TestIntegrationFolderCreatePermissions(t *testing.T) {
 		})
 	}
 }
+
+// TestFoldersCreateAPIEndpointK8S is the counterpart of pkg/api/folder_test.go TestFoldersCreateAPIEndpoint
+func TestFoldersCreateAPIEndpointK8S(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	folderWithoutParentInput := "{ \"uid\": \"uid\", \"title\": \"Folder\"}"
+	folderWithoutUID := "{ \"title\": \"Folder without UID\"}"
+	folderWithTitleEmpty := "{ \"title\": \"\"}"
+	folderWithInvalidUid := "{ \"uid\": \"------------\"}"
+	folderWithUIDTooLong := "{ \"uid\": \"asdfghjklqwertyuiopzxcvbnmasdfghjklqwertyuiopzxcvbnmasdfghjklqwertyuiopzxcvbnm\"}"
+	folderWithSameName := "{\"title\": \"same name\"}"
+
+	type testCase struct {
+		description            string
+		expectedCode           int
+		expectedFolderSvcError error
+		permissions            []resourcepermissions.SetResourcePermissionCommand
+		input                  string
+		createSecondRecord     bool
+	}
+
+	folderCreatePermission := []resourcepermissions.SetResourcePermissionCommand{
+		{
+			Actions:           []string{"folders:create"},
+			Resource:          "folders",
+			ResourceAttribute: "uid",
+			ResourceID:        "*",
+		},
+	}
+
+	// NOTE: folder creation does not return ErrFolderAccessDenied neither ErrFolderNotFound
+	tcs := []testCase{
+		{
+			description:  "folder creation succeeds given the correct request for creating a folder",
```

Comment:
Just to confirm, creating subfolders is not possible, correct?

Reply:
Creating subfolders is currently working both in Legacy and Unified Storage.

Why?

Reply:
would it be better to add an additional test case for it as well? Currently we are only testing the happy path for `folderWithoutParentInput`

Reply:
The idea for this test is just to be a counterpart of [TestFoldersCreateAPIEndpoint](https://github.com/grafana/grafana/blob/b68b69c2b423d6057851f5839fc2d39c397dc069/pkg/api/folder_test.go#L34)

If we are going to add a testcase for the creation of nested folders, then we could do it in a follow-up PR. What do you think?